Realm/RLMArray.mm

```diff
@@ -509,10 +509,14 @@ - (NSUInteger)indexOfObjectWithPredicate:(NSPredicate *)predicate {
 }
 
 - (NSArray *)objectsAtIndexes:(NSIndexSet *)indexes {
-    if (!_backingCollection) {
-        _backingCollection = [NSMutableArray new];
+    try {
+        if (!_backingCollection) {
+            _backingCollection = [NSMutableArray new];
+        }
+        return [_backingCollection objectsAtIndexes:indexes];
+    } catch(...) {
```

Comment:
Why is this swallowing errors? If I remove this and make the managed version throw on out-of-bounds indexing the only tests which fail are things which should be failing.

Reply:
The managed type causes problems with KVO if it throws so I didn’t see a point in letting it throw anywhere at all

Reply:
It appears to work fine to me. The only KVO test which fails is showing that it fixes a place where RLMManagedArray incorrectly differed from RLMArray and NSArray.

Reply:
Oh, it might actually be making a currently weird thing broken in a different way rather than fixing something. Either way, the fix is probably just to validate argument bounds earlier in the process?

Reply:
Changed to not swallow the exception